### PR TITLE
Fix AttributeError in checkout

### LINKED ISSUE #998: Exception Type: AttributeError when trying to create a password after succeful payment
Got this exception:
`Exception Type: AttributeError at /order/5d555f01-0f99-4821-b52f-98fa48efce4b/create-password/
Exception Value: 'NoneType' object has no attribute 'addresses'`

### PR DESCRIPTION
**Pull request by szewczykmira**
Fix #998

### CODE REVIEW

**Comment by patrys** [comment]
It would be nice to know why we no longer try to auth the user.

**Comment by szewczykmira** [comment]
User is authenticated in form's save method - only when we pass request as an argument. We don't need to do it twice.

**Comment by patrys** [comment]
Ok, another question then: why do we have a form that accepts `request`? I think forms are part of the model layer and should not cross into the view/controller territory.

**Comment by szewczykmira** [comment]
It was logic partially taken from previous version of `SignupForm` (from allauth). Do you want me to remove authentication from form's save method?

**Comment by patrys** [comment]
Ideally, yes, I'd like to avoid passing `request` to forms.

File: saleor/order/views.py [hunk 1]
```diff
@@ -141,12 +141,12 @@ def create_password(request, token):
...
-        register_form.save(request)
-        password = form_data.get('password1')
-        auth_user = auth.authenticate(email=email, password=password)
-        if auth_user is not None:
-            auth.login(request, auth_user)
-        attach_order_to_user(order, auth_user)
+        register_form.save()
+        password = form_data.get('password')
+        user = auth.authenticate(email=email, password=password)
+        if user:
```
Comment (patrys) [inline saleor/order/views.py]: Under what circumstances can we not get an authenticated user here? Shouldn't that result in a form error? How do we attach the order if the user is `None`?

Comment (szewczykmira) [inline saleor/order/views.py]: You're right - user won't be authenticated only when there is an issue with form. I'll remove `if user:` in a minute.

File: saleor/order/views.py [hunk 2]
```diff
@@ -141,12 +141,11 @@ def create_password(request, token):
...
         login_form = None
     if register_form.is_valid():
-        register_form.save(request)
-        password = form_data.get('password1')
-        auth_user = auth.authenticate(email=email, password=password)
-        if auth_user is not None:
-            auth.login(request, auth_user)
-        attach_order_to_user(order, auth_user)
+        register_form.save()
+        password = form_data.get('password')
```
Comment (patrys) [inline saleor/order/views.py]: Shouldn't we take the password from `register_form.cleaned_data`?

**Pull request merged**